mavros/src/plugins/setpoint_position.cpp

```diff
@@ -258,6 +282,52 @@ class SetpointPositionPlugin : public plugin::PluginBase,
 		res.success = true;
 		return true;
 	}
+
+	/* -*- rx handler -*- */
+
+	/**
+	 * @brief handle SET_POSITION_TARGET_GLOBAL_INT mavlink msg
```

Comment:
`SET_` prefixes are usually used for messages you send to the FCU, not something you receive from. I understand that what you are doing is basically expect that the data coming from the GCS gets parsed here, since what MAVROS does is nothing more than serve as a proxy to forward messages, but we should not mess how we process data that comes from the FCU and the data that comes from the GCS. This last one should be handled separately.

If the data comes from the FCU, it needs to come in [`POSITION_TARGET_GLOBAL_INT`](https://mavlink.io/en/messages/common.html#POSITION_TARGET_GLOBAL_INT) messages.

Reply:
Btw, for specific GCS data:
- https://github.com/mavlink/mavros/projects/2: still empty but can this feature can be added as an extension to it;
- https://github.com/mavlink/mavros/pull/1117: gives a good example on how to separate streams.

Reply:
@TSC21, thanks for the feedback and links.  I see what you're saying about keeping the handling of messages from the GCS separate from the vehicle.  I was planning to actually make it possible for the vehicle to send set-position-target-local-ned messages so that ROS's navigation can help the vehicle get around some obstacles close to the vehicle.

In any case, I'll think about it a bit more and I'm still trying to find the time to resolve the frame conversion issue in this PR.